internal/store/static.go

```diff
@@ -119,30 +119,35 @@ func (store *StaticStore) getPlugins(serverVersion string, includeEnterprisePlug
 	plugins := map[string]*model.Plugin{}
 
 	for _, storePlugin := range store.plugins {
-		if serverVersion != "" {
-			sv, err := semver.Parse(serverVersion)
-			if err != nil {
-				return nil, errors.Wrapf(err, "failed to parse serverVersion %s", serverVersion)
-			}
-
-			// Workaround for https://mattermost.atlassian.net/browse/MM-26507
-			if sv.GE(semver.MustParse("5.25.0")) && storePlugin.Enterprise && !includeEnterprisePlugins {
+		if storePlugin.Enterprise && !includeEnterprisePlugins {
+			if serverVersion == "" {
 				continue
-			}
-
-			if storePlugin.Manifest.MinServerVersion != "" {
-				var meetsMinServerVersion bool
-				meetsMinServerVersion, err = storePlugin.Manifest.MeetMinServerVersion(serverVersion)
+			} else {
+				sv, err := semver.Parse(serverVersion)
 				if err != nil {
-					return nil, errors.Wrapf(err, "failed to check minServerVersion for manifest.Id %s", storePlugin.Manifest.Id)
+					return nil, errors.Wrapf(err, "failed to parse serverVersion %s", serverVersion)
 				}
 
-				if !meetsMinServerVersion {
+				// Honor enterprise flag for server version >= 5.25.0 only.
+				// Workaround for https://mattermost.atlassian.net/browse/MM-26507
+				minVersionSupportingEnterpriseFlags := semver.MustParse("5.25.0")
```

Comment:
nit: can we define this at the top of the file for clarity, vs. running it each time for each plugin? (Note, it's not a performance concern, just clarity.)

Reply:
I'm not sure, if that would help with clarity. To me a global variable indicates that it's used in multiple methods, which is not the case here.

How would moving this into a global scope make the code more clear?

Reply:
I don't think it's strictly necessary for a global variable to be used in multiple functions, but I do see your point. It's just /definitely/ weird to me have to keep declaring `minVersionSupportingEnterpriseFlags` on each iteration of the loop. Pulling this to the top of the file neatly decouples this, and likely fits in with other such minimum versions that will inevitably arise.

Reply:
Good point on the potential other min versions. I did move it on top.